specification.md

```diff
@@ -5157,43 +5225,37 @@ Workflow constants should not have access to [Workflow secrets definitions](#Wor
 
 ### Workflow Secrets
 
-Secrets allow you access sensitive information, such as passwords, OAuth tokens, ssh keys, etc
-inside your [Workflow Expressions](#Workflow-Expressions). 
+Secrets allow you access sensitive information, such as passwords, OAuth tokens, ssh keys, etc.
+They can be used in [Workflow Expressions](#Workflow-Expressions).
 
 You can define the names of secrets via the [Workflow top-level "secrets" property](#Workflow-Definition-Structure), 
 for example:
 
 ```json
-"secrets": ["MY_PASSWORD", "MY_STORAGE_KEY", "MY_ACCOUNT"]
+"secrets": ["MY_ACCOUNT", "MY_PASSWORD", "urn:example:tenant:accessToken"]
 ```
 
-If secrets are defined in a Workflow definition, runtimes must assure to provide their values
-during Workflow execution.
+If secrets are only named in a Workflow definition, runtimes must assure to 
+provide their values during execution.
 
-Secrets can be used only in [Workflow expressions](#Workflow-Expressions) under the `SECRETS` namespace. 
-This is reserved namespace that should only be allowed for values defined by the `secrets` property.
-
-Here is an example on how to use secrets and pass them as arguments to a function invocation:
+Secrets can be used only in [workflow expressions](#Workflow-Expressions).
+The default expression language JQ provides secrets through the reserved 
+`$SECRETS` variable. Here is an example on how to use secrets and pass them as 
+arguments to a function invocation:
 
 ```json
-"secrets": ["AZURE_STORAGE_ACCOUNT", "AZURE_STORAGE_KEY"],
-
-...
-        
 {
   "refName": "uploadToAzure",
     "arguments": {
-      "account": "${ $SECRETS.AZURE_STORAGE_ACCOUNT }",
-      "account-key": "${ $SECRETS.AZURE_STORAGE_KEY }",
-      ...
+      "user": "${ $SECRETS.MY_ACCOUNT }",
+      "pass": "${ $SECRETS.MY_PASSWORD }",
+      "token": "${ $SECRETS.\"urn:example:tenant:accessToken\" }"
     }
-  
 }
 ```
 
 Note that secrets can also be used in [expression functions](#Using-Functions-for-Expression-Evaluation).
 
-Secrets are immutable, meaning that workflow expressions are not allowed to change their values.
```

Comment:
I think this should maybe be a separate PR if you wish to remove it. 
We decided in our meeting that this should be added when we added secretes/constants. 

I would suggest open a separate pr and define reason for removal please.

Reply:
Sorry, I thought I had moved the text about immutability elsewhere. Or probably because they are passed to the JQ execution as a predefined variable that is not extracted after the interpreter has finished, it can only be immutable. I'll add it back, so that it's clear to the user that secrets can't be set.

Well, technically, the variable can be overwritten during the JQ execution, so it can be "set", it only doesn't propagate back. hmm, maybe neglecting the details and just saying it's immutable is enough for the user to know